src/sentry/grouping/enhancer/__init__.py

```diff
@@ -138,19 +147,28 @@ def apply_modifications_to_frame(
         """
         in_memory_cache: dict[str, str] = {}
 
-        match_frames = [create_match_frame(frame, platform) for frame in frames]
-
-        with sentry_sdk.start_span(
-            op="stacktrace_processing",
-            description="apply_rules_to_frames",
-        ):
+        match_frames, stacktrace_fingerprint = _matching_frames_and_fingerprint(frames, platform)
```

Comment:
This and the other functions I have added have internal `try/except` blocks to prevent any hashing or caching code to abort saving events.